qiskit/transpiler/passes/routing/utils.py

```diff
@@ -0,0 +1,98 @@
+                F(q_i,q_j) = 1 - E(q_i,q_j)
+            The execution time of a swap gate
+                T(SWAP.q_i,SWAP.q_j) = T(q_i,q_j) + T(q_j,q_i) + min(T(q_i,q_j),T(q_j,q_i))
+            The weighted distance matrix
+                D = alpha_dis * swap_distance_matrix +
+                    alpha_error * swap_error_rate_matrix +
+                    alpha_time * swap_execution_time_matrix
+
+    Returns:
+        ndarray: 2D array of distances.
+    """
+    edges = coupling_map.get_edges()
+    two_qubit_gates = []
+    for gate in properties.gates:
+        if len(gate.qubits) == 2:
+            two_qubit_gates.append(gate)
+    error_weights = []
+    execution_time_weights = []
+    cnot_fidelity = cnot_execution_time = reverse_cnot_fidelity = reverse_cnot_execution_time = 0
+    for edge in edges:
+        for gate in two_qubit_gates:
+            if gate.qubits[0] == edge[0] and gate.qubits[1] == edge[1]:
+                cnot_fidelity = 1 - gate.parameters[0].value
+                cnot_execution_time = gate.parameters[1].value
+                break
+        for gate in two_qubit_gates:
+            if gate.qubits[1] == edge[0] and gate.qubits[0] == edge[1]:
+                reverse_cnot_fidelity = 1 - gate.parameters[0].value
+                reverse_cnot_execution_time = gate.parameters[1].value
+                break
+
+        error_weights.append(1 - cnot_fidelity * reverse_cnot_fidelity
+                             * max(cnot_fidelity, reverse_cnot_fidelity))
+        execution_time_weights.append(cnot_execution_time + reverse_cnot_execution_time
+                                      + min(cnot_execution_time, reverse_cnot_execution_time))
+
+    error_weights = np.asarray(error_weights)
+    execution_time_weights = np.asarray(execution_time_weights)
+    coupling_map._compute_distance_matrix()
+    swap_distance_weights = coupling_map._dist_matrix
+    # normalize edge weights
+    avg_error_weight = np.mean(error_weights[error_weights != 1])
+    avg_execution_time_weight = np.mean(execution_time_weights[execution_time_weights != 1])
+    avg_swap_distance_weight = np.linalg.norm(swap_distance_weights)
+
+    normed_error_weights = error_weights / avg_error_weight
+    normed_execution_time_weights = execution_time_weights / avg_execution_time_weight
+    normed_swap_distance_weight = swap_distance_weights / avg_swap_distance_weight
+
+    normed_weights = alpha_error * normed_error_weights + alpha_time * normed_execution_time_weights
+
+    rows = [edge[0] for edge in edges]
+    cols = [edge[1] for edge in edges]
+    weighted_dist = sp.coo_matrix((normed_weights, (rows, cols)),
+                                  shape=(num_qubits, num_qubits),
+                                  dtype=float).tocsr()
+    weighted_paths = cs.shortest_path(weighted_dist, directed=False,
+                                      return_predecessors=False)
```

Comment:
There's usually no need to do `scipy.sparse.coo_matrix(...).tocsr()`, because the CSR constructor should understand the COO triple anyway.  Is there a reason to convert it to CSR and not just let `csgraph.shortest_path` decide what it wants to deal with, anyway?